plugins/image-prioritizer/class-image-prioritizer-img-tag-visitor.php

```diff
@@ -20,70 +20,143 @@
 final class Image_Prioritizer_Img_Tag_Visitor extends Image_Prioritizer_Tag_Visitor {
 
 	/**
-	 * Flag to indicate whether we're within a `<picture>` element.
+	 * Visits a tag.
+	 *
+	 * @since n.e.x.t Separate the processing of <img> and <picture> elements.
+	 * @since 0.1.0
+	 *
+	 * @param OD_Tag_Visitor_Context $context Tag visitor context.
 	 *
-	 * @var bool
+	 * @return bool Whether the tag should be tracked in URL Metrics.
 	 */
-	private $is_within_picture = false;
+	public function __invoke( OD_Tag_Visitor_Context $context ): bool {
+		$processor = $context->processor;
+		$tag       = $processor->get_tag();
+
+		if ( 'PICTURE' === $tag ) {
+			$this->process_picture( $processor, $context );
+			return false;
```

Comment:
This would need to return `true` because we need to capture this tag in URL Metrics, correct? Or maybe we don't if the browser reports that the IMG inside of the PICTURE is the LCP element. We should double check which element is reported as LCP.

Reply:
Yes its need to be true now the `process_picture` return will handle should URL Metrics should be tracked or not.